src/AspNetCoreModuleV2/AspNetCore/src/aspnetcore_shim_config.cpp

```diff
@@ -7,63 +7,36 @@
 #include "hostfxr_utility.h"
 #include "ahutil.h"
 
-ASPNETCORE_SHIM_CONFIG::~ASPNETCORE_SHIM_CONFIG()
-{
-}
-
 HRESULT
 ASPNETCORE_SHIM_CONFIG::Populate(
     IHttpServer      *pHttpServer,
     IHttpApplication *pHttpApplication
 )
 {
-    STACK_STRU(strHostingModel, 300);
-    HRESULT                         hr = S_OK;
+    STACK_STRU(strHostingModel, 10);
     STRU                            strApplicationFullPath;
     IAppHostAdminManager           *pAdminManager = NULL;
-    IAppHostElement                *pAspNetCoreElement = NULL;
-    BSTR                            bstrAspNetCoreSection = NULL;
+    CComPtr<IAppHostElement>        pAspNetCoreElement;
 
     pAdminManager = pHttpServer->GetAdminManager();
-    hr = m_struConfigPath.Copy(pHttpApplication->GetAppConfigPath());
-    if (FAILED(hr))
-    {
-        goto Finished;
-    }
+    RETURN_IF_FAILED(m_struConfigPath.Copy(pHttpApplication->GetAppConfigPath()));
+    RETURN_IF_FAILED(m_struApplicationPhysicalPath.Copy(pHttpApplication->GetApplicationPhysicalPath()));
 
-    hr = m_struApplicationPhysicalPath.Copy(pHttpApplication->GetApplicationPhysicalPath());
-    if (FAILED(hr))
-    {
-        goto Finished;
-    }
-
-    bstrAspNetCoreSection = SysAllocString(CS_ASPNETCORE_SECTION);
+    const CComBSTR bstrAspNetCoreSection = CS_ASPNETCORE_SECTION;
 
-    hr = pAdminManager->GetAdminSection(bstrAspNetCoreSection,
+    RETURN_IF_FAILED(pAdminManager->GetAdminSection(bstrAspNetCoreSection,
         m_struConfigPath.QueryStr(),
-        &pAspNetCoreElement);
-    if (FAILED(hr))
-    {
-        goto Finished;
-    }
+        &pAspNetCoreElement));
 
-    hr = GetElementStringProperty(pAspNetCoreElement,
+    RETURN_IF_FAILED(GetElementStringProperty(pAspNetCoreElement,
         CS_ASPNETCORE_PROCESS_EXE_PATH,
-        &m_struProcessPath);
-    if (FAILED(hr))
-    {
-        goto Finished;
-    }
+        &m_struProcessPath));
 
-    hr = GetElementStringProperty(pAspNetCoreElement,
+    // Swallow this error for backward compatability
+    // Use default behavior for empty string
+    GetElementStringProperty(pAspNetCoreElement,
```

Comment:
Remove this comment. I believe this is incorrect as it will fail to obtain the aspnetcore element.

Reply:
It is correct, it's for hostingModel

Reply:
this is about the scenario that user has very old ANCM schema (before hostingModel was introduced) and new ancm bits. This only can happen in our test scenario

Reply:
No, I think behavior described is incorrect. If someone specified the hostingmodel and they didn't have the schema, I believe it would fail to get the aspnetcore element rather than the hosting model attribute.

Reply:
At this line, we already has the aspnetcore element. the very old V1 schema does aspnetcore element but not hostingModel attribute. this code was inherited from V1.